Sources/CommonMark/Visitor.swift

```diff
@@ -0,0 +1,387 @@
+    }
+
+    public func visit(htmlBlock: HTMLBlock) -> VisitorContinueKind {
+        return .inherit
+    }
+
+    public func visitPost(htmlBlock: HTMLBlock) {
+        // nothing by default
+    }
+
+    public func visit(codeBlock: CodeBlock) -> VisitorContinueKind {
+        return .inherit
+    }
+
+    public func visitPost(codeBlock: CodeBlock) {
+        // nothing by default
+    }
+
+    public func visit(thematicBreak: ThematicBreak) -> VisitorContinueKind {
+        return .inherit
+    }
+
+    public func visitPost(thematicBreak: ThematicBreak) {
+        // nothing by default
+    }
+
+    // MARK: - Inline
+
+    public func visit(inline: Inline) -> VisitorContinueKind {
+        return .inherit
+    }
+
+    public func visitPost(inline: Inline) {
+        // nothing by default
+    }
+
+    public func visit(text: Text) -> VisitorContinueKind {
+        return .inherit
+    }
+
+    public func visitPost(text: Text) {
+        // nothing by default
+    }
+
+    public func visit(strong: Strong) -> VisitorContinueKind {
+        return .inherit
+    }
+
+    public func visitPost(strong: Strong) {
+        // nothing by default
+    }
+
+    public func visit(emphasis: Emphasis) -> VisitorContinueKind {
+        return .inherit
+    }
+
+    public func visitPost(emphasis: Emphasis) {
+        // nothing by default
+    }
+
+    public func visit(link: Link) -> VisitorContinueKind {
+        return .inherit
+    }
+
+    public func visitPost(link: Link) {
+        // nothing by default
+    }
+
+    public func visit(image: Image) -> VisitorContinueKind {
+        return .inherit
+    }
+
+    public func visitPost(image: Image) {
+        // nothing by default
+    }
+
+    public func visit(code: Code) -> VisitorContinueKind {
+        return .inherit
+    }
+
+    public func visitPost(code: Code) {
+        // nothing by default
+    }
+
+    public func visit(rawHTML: RawHTML) -> VisitorContinueKind {
+        return .inherit
+    }
+
+    public func visitPost(rawHTML: RawHTML) {
+        // nothing by default
+    }
+
+    public func visit(softLineBreak: SoftLineBreak) -> VisitorContinueKind {
+        return .inherit
+    }
+
+    public func visitPost(softLineBreak: SoftLineBreak) {
+        // nothing by default
+    }
+
+    public func visit(hardLineBreak: HardLineBreak) -> VisitorContinueKind {
+        return .inherit
+    }
+
+    public func visitPost(hardLineBreak: HardLineBreak) {
+        // nothing by default
+    }
+}
+
+extension Visitor {
+    internal func visit<T: Visitable>(
+        _ visitable: T,
+        by visitLeafTypeOf: @autoclosure () -> VisitorContinueKind
+    ) -> VisitorContinueKind {
+        let userDefaultContinueKind = self.defaultContinueKind
+
+        let inheritedContinueKind = self.visitSuperTypesOf(visitable: visitable)
+        let leafTypeContinueKind = visitLeafTypeOf()
+
+        // Abort on debug:
+        assert(
+            userDefaultContinueKind != .inherit,
```

Comment:
Assertions like this are a code smell for me; illegal states should be avoided through the type system. As per my comment above, could we get rid of user-configurable `defaultContinueKind` and instead say that the default behavior is `.visitChildren` (overridable by `visit(node:)`)?

Reply:
Fixed!